Clarify steps in managing open pull requests

jasonblais opened the pull request.

lfbrock commented on source/developer/contribution-guide.md:
```diff
@@ -53,16 +53,23 @@ After submitting a PR, before it is merged:
-If you've included your mailing address in Step 1, you may receive a [Limited Edition Mattermost Mug](https://forum.mattermost.org/t/limited-edition-mattermost-mugs/143) as a thank you gift after your first pull request is merged. 
+2. PM review must pass
+    - A Product Manager will review the pull request to make sure it:
+        - fits with our product roadmap
```
Can you capitalize these bullet points?

lfbrock commented on source/developer/contribution-guide.md:
```diff
@@ -53,16 +53,23 @@ After submitting a PR, before it is merged:
+        - fits with our product roadmap
+        - works as expected
+        - meets [UX guidelines](https://docs.mattermost.com/developer/fx-guidelines.html)
+    - The Product Manager may come back with some bugs or UI improvements to fix before the pull request moves on to dev review.
```
Can you update the bullet points to all be consistent with period or no period?

lfbrock commented on source/developer/contribution-guide.md:
```diff
@@ -53,16 +53,23 @@ After submitting a PR, before it is merged:
 _Please DO NOT @-mention reviewers if they haven't yet commented on your PR._
 _Pull requests are reviewed in sequence._
 
 **Approved reviewers include**: coreyhulen, crspeller, hmhealey, jwilander  
```
We should also update these lists to add George + Elias

jasonblais replied:
Right, I meant to add them... thanks for the notice

lfbrock commented:
+1

I think we're still a little inconsistent in this doc with combining fragments and sentences in the same list, but not sure if there's an easy way to make it better.

esethna commented on source/developer/contribution-guide.md:
```diff
@@ -42,27 +42,34 @@ Before submitting a pull request (PR), check that:
+
+If you've included your mailing address in the signed [Contributor License Agreement](https://www.mattermost.org/mattermost-contributor-agreement/), you may receive a [Limited Edition Mattermost Mug](https://forum.mattermost.org/t/limited-edition-mattermost-mugs/143) as a thank you gift after your first pull request is merged. 
 
 ### Approved Reviewers
```
We should probably split this out into approved dev and approved PM reviewers.

esethna commented on source/developer/contribution-guide.md:
```diff
@@ -42,27 +42,34 @@ Before submitting a pull request (PR), check that:
 
 ### Approved Reviewers
 
 _Please DO NOT @-mention reviewers if they haven't yet commented on your PR._
```
This porbably doesn't need to be italicized, it looks kinda funky

jasonblais commented:
1. Removed the "do not @mention" note as I feel that's unnecessary.
2. Added approved PM reviewers
3. Removed seconders, which is no longer applicable for us
